Experimental player textures & logs!

Liam0 opened the pull request:
Added logging to all admin actions.

Logging is using a Database function instead of PublicVariableEventHandler(it can still print to rpt if u have any reason to do so)

Changed the getID button in the admin menu to display the log dialog showing admin and player logs, initial commits actually had this in the commit messages but i removed those when i squashed the commits together just to make it nice and clean for anyone who has to read/implement it. it should now be displayed alongside the rest of the playerinfo within the admin menu.

Is there even anyone using getID??

Player textures converted to use a config file ie.
{ "textures\cop_uniform.jpg", "life_coplevel", "SCALAR", 1, "cop"},

Additional fixes

Typo in localize string, fn_restrain.sqf
Fix to stop vinteraction menu from opening while restrained.
Fix to stop players from entering turrets on vehicles while being restrained.

sargken commented on Altis_Life.Altis/Config_Master.hpp:
should only be 1 tab for indentation but that's just my opinion.

sargken commented on Altis_Life.Altis/Config_Master.hpp:
Should not leave tabs in empty lines.

Liam0 commented:
@sargken there u go.

Liam0 commented:
i didn't localize because i know @Jawshy is working on localisation right now and i dont want to screw him over with conflicts.

tkcjesse commented:
Won't logging this way increase server overhead substantially?

Liam0 commented:
@tkcjesse, yeah i'll just remove that, ur right it shouldn't be necessary. And that depends on how much stuff u want to log i guess. Hopefully it doesn't.

tkcjesse commented:
Also, Why'd you remove getID?

Liam0 commented:
it's still there..

tkcjesse commented on Altis_Life.Altis/core/admin/fn_admingetID.sqf:
```diff
@@ -1,16 +0,0 @@
-/*
-	File: fn_admingetID.sqf
```
@Liam0 Regarding how you said getID was still here? Help me out. What am I missing?

tkcjesse commented on Altis_Life.Altis/Functions.hpp:
```diff
@@ -68,16 +68,15 @@ class Life_Client_Core {
 		class adminCompensate {};
 		class adminDebugCon {};
 		class adminFreeze {};
-		class admingetID {};
```
@Liam0 Regarding how you said getID was still here? Help me out. What am I missing?

tkcjesse commented on Altis_Life.Altis/CfgRemoteExec.hpp:
```diff
@@ -88,10 +89,10 @@ class CfgRemoteExec {
         F(TON_fnc_chopShopSell,SERVER)
         F(TON_fnc_cleanupRequest,SERVER)
         F(TON_fnc_deleteDBContainer,SERVER)
-        F(TON_fnc_getID,SERVER)
```
@Liam0 Regarding how you said getID was still here? Help me out. What am I missing?

tkcjesse commented on Altis_Life.Altis/dialog/admin_menu.hpp:
```diff
@@ -39,8 +39,8 @@ class life_admin_menu {
 
 		class RscButtonMenu_2402: Life_RscButtonMenu {
 			idc = -1;
-			text = "$STR_Admin_GetID";
```
@Liam0 Regarding how you said getID was still here? Help me out. What am I missing?

tkcjesse commented:
@Liam0 Help me out, I see where you removed & wrote over getID, but where is it now? I don't see it. Help me out please.

Liam0 commented:
@tkcjesse fn_admininfo.sqf

tkcjesse commented:
Did I try it? No. Can't be bothered reading every single change in 49 files... I read it yes, briefly. Considering I obviously noticed it was removed and changed. Sorry I didn't notice 2 extremely small changes to a single file that has the changes to where I need to scroll to the right. In 49 changed files. HENCE why I asked you to please help me out. It wasn't meant rude or anything. Just didn't see it anywhere, considering your "ChangeLog" isn't exactly detailed to explain what you did with it. Hell, it's not even mentioned.

Liam0 commented:
Im not asking you to read every single addition but maybe read the relevant ones if there is something u dont understand. i can't spend more time on the changelog then the actual files

But here you go just because you asked so nicely.

Added logging to all admin actions.

Logging is using a Database function instead of PublicVariableEventHandler(it can still print to rpt if u have any reason to do so)

Changed the getID button in the admin menu to display the log dialog showing admin and player logs, initial commits actually had this in the commit messages but i removed those when i squashed the commits together just to make it nice and clean for anyone who has to read/implement it. it should now be displayed alongside the rest of the playerinfo within the admin menu.

Is there even anyone using getID??

Player textures converted to use a config file ie.
{ "textures\cop_uniform.jpg", "life_coplevel", "SCALAR", 1, "cop"},

Additional fixes

Typo in localize string, fn_restrain.sqf
Fix to stop vinteraction menu from opening while restrained.
Fix to stop players from entering turrets on vehicles while being restrained.

it's got some merge conflicts after the last few merges that i need to fix later because i dont have time right now.

tkcjesse commented:
I've already brought up removing GetID completely. Yes people use it. Those who don't use a remote admin tool like DaRT. It's for in-game bans and kicks.

BoGuu commented on life_server/Functions/Systems/fn_logIt.sqf:
```diff
@@ -1,8 +1,39 @@
+if(_admin) then {
+
+	if(EQUAL(LIFE_SETTINGS(getNumber,"player_rptlog"),1)) then {
+		diag_log format["[LOG] ADMIN %1", _log];
```
_log isn't defined yet

Liam0 replied:
Thanks
